image-encapsulate.cpp

```diff
@@ -1,19 +1,24 @@
 #include <iostream>
 #include "brightener.h"
 
-int main() {
-	auto image = std::make_unique<Image>(512, 512);
+int main() 
+{
+	auto image = std::make_shared<Image>(512, 512);
 	std::cout << "Brightening a 512 x 512 image\n";
-	ImageBrightener brightener(std::move(image));
-	if (brightener.ValidateImage()) {
+	std::cout << image.use_count() << std::endl;
+
+	if (image->ValidateImage())
```

Comment:
Nice move of `ValidateImage` to the `Image` itself. Now it's possible to avoid initializing the `ImageBrightener` if the image isn't valid 👍